### PR TITLE
PT-2048 - pt-osc spans excessive connections to the replica when executing in the source

### PR DESCRIPTION
**Pull request by svetasmirnova**
The bug itself is fixed by PT-2160: 'fix tests for pt online schema change'

This PR adds a regression test.

- [x] The contributed code is licensed under GPL v2.0
- [x] Contributor Licence Agreement (CLA) is signed
- [ ] util/update-modules has been ran
- [ ] Documention updated
- [x] Test suite update

### CODE REVIEW

**Review by EvgeniyPatlan — APPROVED** [review]

File: t/pt-online-schema-change/pt-2048.t [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    unshift @INC, "$ENV{PERCONA_TOOLKIT_BRANCH}/lib";
+};
+
+use strict;
+use warnings FATAL => 'all';
+
+use English qw(-no_match_vars);
+use Test::More;
+
+use Data::Dumper;
```
Comment (saikumar-vs) [inline t/pt-online-schema-change/pt-2048.t]: Not a must, but we can remove Dumper

File: t/pt-online-schema-change/pt-2048.t [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+use PerconaTest;
+use Sandbox;
+
+require "$trunk/bin/pt-online-schema-change";
+
+my $dp = new DSNParser(opts=>$dsn_opts);
+my $sb = new Sandbox(basedir => '/tmp', DSNParser => $dp);
+
+my $dbh = $sb->get_dbh_for('master');
+my $slave_dbh  = $sb->get_dbh_for('slave1');
```
Comment (saikumar-vs) [inline t/pt-online-schema-change/pt-2048.t]: extra space before assignment operator

File: t/pt-online-schema-change/pt-2048.t [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+my $dsn = $sb->dsn_for("master");
```
Comment (saikumar-vs) [inline t/pt-online-schema-change/pt-2048.t:25]: an improvement suggestion:
if ( !dbh ) {
   plan skip_all => 'Cannot connect to sandbox master';
}
elsif ( !$slave_dbh ) {
   plan skip_all => 'Cannot connect to sandbox slave';
}

**Review by saikumar-vs — COMMENTED** [review]
LGTM, Thanks for working on this.

Few cosmetic and improvements suggested, but dosent block the push.

**Review by saikumar-vs — APPROVED** [review]

**Pull request merged**